app/js/user.js

```diff
@@ -88,6 +87,49 @@
           });
         });
       });
+    },
+
+    /**
+     * Updates the local user data as well as last sync time.
+     * @param {object} user_data - The ammended user data and settings i.e.
+     * user_data = {
+     *   user: user,
+     *   settings, settings
+     * };
+     * @returns The updated user object from indexedDB
+     */
+    update_user: function(user_data) {
+      SumoDB.update_user(user_data).then(function(user) {
+        user.last_sync = Date.now();
+        return asyncStorage.setItem('user', user).then(function() {
```

Comment:
The hardcoded values were just hardcoded temporarily, while we waited for this bug. We want to save all the fields in the profile on the server so that a helper can log into another device and have the same settings.
So yes, we need to normalize. And save the handset type and operator on the server.